src/utils/evm/parsing.ts

```diff
@@ -0,0 +1,44 @@
+import { ethers } from 'ethers'
+import { parseMethod } from '../parsing'
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export function extractContractAddress(rawData: Record<string, any>) {
+  const [firstParameter] = rawData.params
+  const method = parseMethod(rawData)
+
+  if (method === 'eth_sendRawTransaction') {
```

Comment:
> you could change this to a switch

Valid! I did first do it with a switch, for some reason I did not use it because the way I was parsing the methods. That got replaced shortly and I didn't turn this back to a switch.

Replaced now!